### PR TITLE
[PF-2768] Handle deserializing placeholder gcp contexts

### PR DESCRIPTION
**Pull request by zloery**
Currently, calling `createCloudContextStart` puts a placeholder value in the cloud_context DB. This is important for state management, but it does not have any GCP context information (policy groups, project ID) and so deserialization on this placeholder object will fail. IMO we should hide this placeholder as it will not have the fields most callers expect.

This is currently calling `getWorkspace` calls immediately after `createGcpCloudContext` calls to fail with a 400. After the context creation finishes, the `getWorkspace` calls start working again.

### CODE REVIEW

File: service/src/main/java/bio/terra/workspace/service/workspace/model/GcpCloudContext.java [hunk 1]
```diff
@@ -80,14 +81,29 @@ public String serialize() {
...
+   */
+  public static Optional<GcpCloudContext> deserialize(DbCloudContext dbCloudContext) {
     // Sanity check the input
     if (dbCloudContext == null
         || dbCloudContext.getSpendProfile() == null
         || dbCloudContext.getCloudPlatform() != CloudPlatform.GCP) {
       throw new InternalLogicException("Invalid GCP cloud context state");
     }
 
+    if (dbCloudContext.getState() == WsmResourceState.CREATING) {
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/workspace/service/workspace/model/GcpCloudContext.java:103]: or  broken?

File: service/src/test/java/bio/terra/workspace/service/workspace/GcpCloudContextUnitTest.java [hunk 2]
```diff
@@ -92,6 +93,37 @@ void serdesTest() {
...
+            "a-user-facing-id",
+            "deleteGcpContextDeletesControlledResources",
+            "description",
+            new SpendProfileId("spend-profile"),
+            Collections.emptyMap(),
+            WorkspaceStage.MC_WORKSPACE,
+            DEFAULT_USER_EMAIL,
+            null);
+    workspaceDao.createWorkspace(workspace, /* applicationIds= */ null);
+    workspaceDao.createCloudContextStart(
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/workspace/service/workspace/GcpCloudContextUnitTest.java:111]: nit: line break between setup, test, assert

**Review by yuhuyoyo — APPROVED** [review]

**Review by cbookg — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/service/workspace/model/GcpCloudContext.java (hunk 1 above)
Comment (ddietterich) [reply service/src/main/java/bio/terra/workspace/service/workspace/model/GcpCloudContext.java:103]: BROKEN also probably doesn't have the fields filled in. Right now, no one is using BROKEN, so it probably doesn't matter.

**Review by ddietterich — APPROVED** [review]
Fine for now. I will have to make a bigger fix in order to get the target behavior.

**Comment by ddietterich** [comment]
I made [PF-2771](https://broadworkbench.atlassian.net/browse/PF-2771) to track the proper fix

**Pull request merged**